cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/monitor/sampling/CruiseControlMetricsReporterSampler.java

```diff
@@ -49,80 +47,71 @@
   private static final Random RANDOM = new Random();
 
   private Consumer<String, CruiseControlMetric> _metricConsumer;
+  private String _metricReporterTopic;
+  private Set<TopicPartition> _currentPartitionAssignment;
+
   @Override
   public Samples getSamples(Cluster cluster,
                             Set<TopicPartition> assignedPartitions,
                             long startTimeMs,
                             long endTimeMs,
                             SamplingMode mode,
-                            MetricDef metricDef) throws MetricSamplingException {
-    // Ensure we have an assignment.
-    long pollerCount = 0L;
-    while (_metricConsumer.assignment().isEmpty()) {
-      pollerCount++;
-      _metricConsumer.poll(10);
-      if (pollerCount % (12000) == 0) {
-        LOG.warn("metricConsumer Assignment is empty .. Did you copy the cruise-control-metrics-reporter.jar to Kafka libs ?");
-      }
+                            MetricDef metricDef,
+                            long timeout) {
+    // Commit consumer offsets and refresh partition assignment.
+    _metricConsumer.commitSync();
```

Comment:
I am thinking do we need to commit offset here? Since below we always seek to certain offset before we begin to consume.

Reply:
Good point!